pep-0692.rst

```diff
@@ -0,0 +1,438 @@
+already existing codebases where the need of refactoring the code in order to
+introduce proper type annotations may be considered not worth the effort. This
+in turn prevents the project from getting all of the benefits that type hinting
+can provide. As a consequence, there has been a lot of discussion around
+supporting more precise ``**kwargs`` typing [#mypyIssue4441]_ and it became a
+feature that is necessary for a large part of Python community.
+
+Rationale
+=========
+
+PEP 589 introduced the ``TypedDict`` type constructor that supports dictionary
+types consisting of string keys and values of potentially different types. A
+function's keyword arguments represented by a formal parameter that begins with
+double asterisk, such as ``**kwargs``, are received as a dictionary. This makes
+``TypedDict`` a perfect candidate to be used for more precise ``**kwargs``
+typing. In addition, with ``TypedDict`` keyword names can be taken into account
+during static type analysis. However, specifying ``**kwargs`` type with a
+``TypedDict`` means, as mentioned earlier, that each keyword argument specified
+by ``**kwargs`` is a ``TypedDict`` itself. For instance:
+
+::
+
+    class Movie(TypedDict):
+        name: str
+        year: int
+    
+    def foo(**kwargs: Movie) -> None: ...
+
+means that each keyword argument in ``foo`` is itself a ``Movie`` dictionary
+that has a ``name`` key with a string type value and a ``year`` key with an
+integer type value. Therefore, in order to support specifying ``kwargs`` type
+as a ``TypedDict`` without breaking current behaviour, a new syntax has to be
+introduced.
+
+Specification
+=============
+
+To support the aforementioned use case we propose to use the double asterisk
+syntax inside of the type annotation. The required grammar change is discussed
+in more detail in section `Grammar Changes`_. Continuing the previous example:
+
+::
+
+    def foo(**kwargs: **Movie) -> None: ...
+
+would mean that the ``**kwargs`` comprise two keyword arguments specified by
+``Movie`` (i.e. a ``name`` keyword of type ``str`` and a ``year`` keyword of
+type ``int``). Then, inside the function itself, the type checkers should treat
+the ``kwargs`` parameter as a ``TypedDict``:
+
+::
+
+    def foo(**kwargs: **Movie) -> None:
+        assert_type(kwargs, Movie)  # OK!
+
+
+Using the new annotation will not have any runtime effect - it should only be
+taken into account by the type checkers.
+
+Function calls with standard dictionaries
+-----------------------------------------
+
+Calling a function that has ``**kwargs`` typed using the ``**kwargs: **Movie``
+syntax with a standard dictionary must generate an error. For example:
+
+::
+
+    def foo(**kwargs: **Movie) -> None: ...
+
+    movie: dict[str, object] = {"name": "Life of Brian", "year": 1979}
+    foo(**movie)  # WRONG! Movie is of type dict[str, object]
+
+    typedMovie: Movie = {"name": "The Meaning of Life", "year": 1983}
+    foo(**typedMovie)  # OK!
```

Comment:
```suggestion
    typed_movie: Movie = {"name": "The Meaning of Life", "year": 1983}
    foo(**typed_movie)  # OK!
```
Nit: [Camelcase is not for Python.](https://www.youtube.com/watch?v=hgI0p1zf31k)